lib/json.rb

```diff
@@ -2,88 +2,405 @@
+#   that control both input allowed and output formatting.
 #
-# Creating a JSON string for communication or serialization is
-# just as simple.
+# The difference between the two methods
+# is that JSON.parse! omits some checks
+# and may not be safe for some +source+ data;
+# use it only for data from trusted sources.
+# Use the safer method JSON.parse for less trusted sources.
 #
-#   require 'json'
+# ==== Parsing \JSON Arrays
 #
-#   my_hash = {:hello => "goodbye"}
-#   puts JSON.generate(my_hash) # => "{\"hello\":\"goodbye\"}"
+# When +source+ is a \JSON array, JSON.parse by default returns a Ruby \Array:
+#   json = '["foo", 1, 1.0, 2.0e2, true, false, null]'
+#   ruby = JSON.parse(json)
+#   ruby # => ["foo", 1, 1.0, 200.0, true, false, nil]
+#   ruby.class # => Array
 #
-# Or an alternative way:
+# The \JSON array may contain nested arrays, objects, and scalars
+# to any depth:
+#   json = '[{"foo": 0, "bar": 1}, ["baz", 2]]'
+#   JSON.parse(json) # => [{"foo"=>0, "bar"=>1}, ["baz", 2]]
 #
-#   require 'json'
-#   puts({:hello => "goodbye"}.to_json) # => "{\"hello\":\"goodbye\"}"
+# ==== Parsing \JSON \Objects
+#
+# When the source is a \JSON object, JSON.parse by default returns a Ruby \Hash:
+#   json = '{"a": "foo", "b": 1, "c": 1.0, "d": 2.0e2, "e": true, "f": false, "g": null}'
+#   ruby = JSON.parse(json)
+#   ruby # => {"a"=>"foo", "b"=>1, "c"=>1.0, "d"=>200.0, "e"=>true, "f"=>false, "g"=>nil}
+#   ruby.class # => Hash
+#
+# The \JSON object may contain nested arrays, objects, and scalars
+# to any depth:
+#   json = '{"foo": {"bar": 1, "baz": 2}, "bat": [0, 1, 2]}'
+#   JSON.parse(json) # => {"foo"=>{"bar"=>1, "baz"=>2}, "bat"=>[0, 1, 2]}
+#
+# ==== Parsing \JSON Scalars
+#
+# When the source is a \JSON scalar (not an array or object),
+# JSON.parse returns a Ruby scalar.
+#
+# \String:
+#   ruby = JSON.parse('"foo"')
+#   ruby # => "foo"
+#   ruby.class # => String
+# \Integer:
+#   ruby = JSON.parse('1')
+#   ruby # => 1
+#   ruby.class # => Integer
+# \Float:
+#   ruby = JSON.parse('1.0')
+#   ruby # => 1.0
+#   ruby.class # => Float
+#   ruby = JSON.parse('2.0e2')
+#   ruby # => 200
+#   ruby.class # => Float
+# Boolean:
+#   ruby = JSON.parse('true')
+#   ruby # => true
+#   ruby.class # => TrueClass
+#   ruby = JSON.parse('false')
+#   ruby # => false
+#   ruby.class # => FalseClass
+# Null:
+#   ruby = JSON.parse('null')
+#   ruby # => nil
+#   ruby.class # => NilClass
+#
+# === Generating \JSON
+#
+# To generate a Ruby \String containing \JSON data,
+# use method <tt>JSON.generate(source, opts)</tt>, where
+# - +source+ is a Ruby object.
+# - +opts+ is a \Hash object containing options
+#   that control both input allowed and output formatting.
+#
+# ==== Generating \JSON from Arrays
+#
+# When the source is a Ruby \Array, JSON.generate returns
+# a \String containing a \JSON array:
+#   ruby = [0, 's', :foo]
+#   json = JSON.generate(ruby)
+#   json # => "[0,\"s\",\"foo\"]"
```

Comment:
Agree, more legible, but I'm reluctant to do that.  I (almost) always copy the output directly from irb.  The few times I don't is when I make mistakes.  (Should we file an enhancement request for irb?)

Reply:
I use `puts json` and add the single quotes manually.
I created https://bugs.ruby-lang.org/issues/16985
In the meantime, it would definitely improve legibility

Reply:
Awesome. I can't mark this (or other resolved points) as resolved...